homeassistant/components/ping/__init__.py

```diff
@@ -21,43 +21,33 @@
 
 CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.DEVICE_TRACKER, Platform.SENSOR]
-
-
-@dataclass(slots=True)
-class PingDomainData:
-    """Dataclass to store privileged status."""
-
-    privileged: bool | None
+DATA_PRIVILEGED_KEY: HassKey[bool | None] = HassKey(DOMAIN)
 
 
 type PingConfigEntry = ConfigEntry[PingUpdateCoordinator]
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the ping integration."""
-
-    hass.data[DOMAIN] = PingDomainData(
-        privileged=await _can_use_icmp_lib_with_privilege(),
-    )
+    hass.data[DATA_PRIVILEGED_KEY] = await _can_use_icmp_lib_with_privilege()
```

Comment:
Any particular reason this is here and we just don't move it to `setup_entry`?

Reply:
Actually yes, we just need to check once whether we have the rights to open a raw socket or not. If we now move this to the `setup_entry`, we would do this for each entry.